src/Core/Cache/StandardCacheWarmer.php

```diff
@@ -0,0 +1,297 @@
+ * party implementation of FluidCacheWarmerInterface which allows
+ * them to be specific to the framework in which Fluid is used.
+ * The default set of mitigation suggestions are based on the
+ * standard errors which can be thrown by the Fluid engine.
+ */
+class StandardCacheWarmer implements FluidCacheWarmerInterface {
+
+	/**
+	 * Warm up an entire collection of templates based on the
+	 * provided RenderingContext (the TemplatePaths carried by
+	 * the RenderingContext, to be precise).
+	 *
+	 * Returns a FluidCacheWarmupResult with result information
+	 * about all detected template files and the compiling of
+	 * those files. If a template fails to compile or throws an
+	 * error, a mitigation suggestion is included for that file.
+	 *
+	 * @param RenderingContextInterface $renderingContext
+	 * @return FluidCacheWarmupResult
+	 */
+	public function warm(RenderingContextInterface $renderingContext) {
+		$renderingContext->getTemplateCompiler()->enterWarmupMode();
+		$formats = array('html', 'xml', 'txt', 'json', 'rtf', 'atom', 'rss');
+		$result = new FluidCacheWarmupResult();
+		$result->merge(
+			$this->warmupTemplateRootPaths($renderingContext, $formats),
+			$this->warmupPartialRootPaths($renderingContext, $formats),
+			$this->warmupLayoutRootPaths($renderingContext, $formats)
+		);
+		return $result;
+	}
+
+	/**
+	 * Warm up _templateRootPaths_ of the RenderingContext's
+	 * TemplatePaths instance.
+	 *
+	 * Scans for template files recursively in all template root
+	 * paths while respecting overlays, e.g. if a path replaces
+	 * the template file of a lower priority path then only
+	 * one result is returned - the overlayed template file. In
+	 * other words the resolving happens exactly as if you were
+	 * attempting to render each detected controller, so that the
+	 * compiled template will be the same that is resolved when
+	 * rendering that controller.
+	 *
+	 * Also scans the root level of all templateRootPaths for
+	 * controller-less/fallback-action template files, e.g. files
+	 * which would be rendered if a specified controller's action
+	 * template does not exist (fallback-action) or if no controller
+	 * name was specified in the context (controller-less).
+	 *
+	 * Like other methods, returns a FluidCacheWarmupResult instance
+	 * which can be merged with other result instances.
+	 *
+	 * @param RenderingContextInterface $renderingContext
+	 * @param array $formats
+	 * @return FluidCacheWarmupResult
+	 */
+	protected function warmupTemplateRootPaths(RenderingContextInterface $renderingContext, array $formats) {
+		$result = new FluidCacheWarmupResult();
+		$paths = $renderingContext->getTemplatePaths();
+		foreach ($formats as $format) {
+			$paths->setFormat($format);
+			foreach ($this->detectControllerNamesInTemplateRootPaths($paths->getTemplateRootPaths()) as $controllerName) {
+				foreach ($paths->resolveAvailableTemplateFiles($controllerName, $format) as $templateFile) {
```

Comment:
we might want to refactor to reduce nesting here, although it's still "ok" i guess